IPython/utils/module_paths.py

```diff
@@ -26,25 +26,46 @@
 from __future__ import print_function
 
 # Stdlib imports
-import imp
 import os
-
-# Third-party imports
-
-# Our own imports
-
+import sys
 
 #-----------------------------------------------------------------------------
-# Globals and constants
+# Classes and functions
 #-----------------------------------------------------------------------------
 
-#-----------------------------------------------------------------------------
-# Local utilities
-#-----------------------------------------------------------------------------
+import imp
+def _find_module_34(name, path):
+    if name is None:
+        return None
+    try:
+        file, filename, _ = imp.find_module(name, path)
+    except ImportError:
+        return None
+    if file is None:
+        return filename
+    else:
+        file.close()
+    if os.path.splitext(filename)[1] in [".py", ".pyc"]:
+        return filename
+    else:
+        return None
+
+if sys.version_info > (3,4):
```

Comment:
Yes, probably, I have a lot to cleanup. `>=` or `>` don't change much as version_info is a 3+ tuple and `(3,4, <anything>)` will compare higher in all cases:

``` 
In [1]: (3,4) < (3,4, None)
Out[1]: True

In [2]: (3,4) < (3,4, -12)
Out[2]: True
```